py/desispec/scripts/tsnr_ensemble.py

```diff
@@ -0,0 +1,147 @@
+import astropy.io.fits               as     fits
+import desisim.templates
+
+from   astropy.convolution           import convolve, Box1DKernel
+from   pathlib                       import Path
+from   desiutil.dust                 import mwdust_transmission
+from   desiutil.log                  import get_logger
+
+
+np.random.seed(seed=314)
+
+# AR/DK DESI spectra wavelengths
+# TODO:  where are brz extraction wavelengths defined?  https://github.com/desihub/desispec/issues/1006.                                                                                                                              
+wmin, wmax, wdelta = 3600, 9824, 0.8
+wave               = np.round(np.arange(wmin, wmax + wdelta, wdelta), 1)
+cslice             = {"b": slice(0, 2751), "r": slice(2700, 5026), "z": slice(4900, 7781)}
+
+def parse(options=None):
+    parser = argparse.ArgumentParser(description="Generate a sim. template ensemble of given type.")
+    parser.add_argument('--nmodel', type = int, default = 2000, required=True,
+                        help='Number of galaxies in the ensemble.')
+    parser.add_argument('--tracer', type = str, default = 'bgs', required=True,
+                        help='Tracer to generate.')
+    parser.add_argument('--outdir', type = str, default = 'bgs', required=True,
+			help='Directory to write to.')
+    args = None
+
+    if options is None:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(options)
+
+    return args
+        
+class template_ensemble(object):
+    '''                                                                                                                                                                                                                                   
+    Generate an ensemble of templates to sample tSNR for a range of points in                                                                                                                                                             
+    (z, m, OII, etc.) space.                                                                                                                                                                                                                                                                                                                                                                                                                                                       
+    If conditioned, uses deepfield redshifts and (currently r) magnitudes to condition simulated templates.                                                                                                                               
+    '''
+    def __init__(self, outdir, tracer='ELG', nmodel=5):        
+        def tracer_maker(wave, tracer=tracer, nmodel=nmodel, redshifts=None, mags=None):
+            # https://arxiv.org/pdf/1611.00036.pdf
+            if tracer == 'elg':
```

Comment:
The default `template_ensemble` constructor `tracer` is "ELG" (uppercase), while this `tracer_maker` function appears to assume all tracers are lowercase.  Let's be consistent throughout, or otherwise include code like `tracer = tracer.lower()` to make it input case agnostic.